### ISSUE TITLE
Advanced Settings

### ISSUE THREAD

**Issue by wada3n** [giantpinkrobots/varia#30]
I think that since the Backend aria2 supports a lot of things, it should be added to the Preferences, such as:

> -x, --max-connection-per-server, -m, --max-tries, -k, --min-split-size, -s, --split, -t, --timeout ......

I find what is suitable for it in a tab called Advanced in the Preferences window, so that the application maintains its simplicity and at the same time is rich with the richness of aria2.
![Capture d’écran du 2023-12-08 21-06-17](https://github.com/giantpinkrobots/varia/assets/13304792/4de8e716-4bba-4ea9-99a6-3bb16d1759b1)

**Comment by bragefuglseth** [comment]
A lot of these settings seem very niche to me, and I find it hard to see how anyone would find them [worth the cost](https://ometer.com/preferences.html). Are there any cases where this level of tweaking is actually useful?

**Comment by wada3n** [comment]
Hahaha, of course not all features need to be added to the app, but I believe some additional features would make the manager more powerful and user-friendly.
Timeout, File splitting, Multi-server connections, these features are very important, and will speed up the downloading, otherwise why would the user use the app if it does not offer something additional like this.

**Comment by bragefuglseth** [comment]
> these features are very important, and will speed up the downloading

I don’t doubt that they are, but what concrete use cases do they have? Why are they useful, and what problems do they solve?

**Comment by wada3n** [comment]
Simply put, not everyone has fast internet. and use (Split and multi Conn) can enhance and Increase download speed.
In other words, trying to get every drop of the connection available to user, unlike the normal method, one connection may be fast or not.
btw, thank you very much for the link and discussion, it is very useful.
These settings are placed in the advanced tab because they may add problems in some servers that prevent multiple connections, and you may get a corrupted file in the end, so they are often treated with caution, so these additions may actually be added for advanced users, and the average user cannot use them, or rather, they may not need them. Implicitly, for example, adding a proxy, a cookie file, this is important and has benefit in downloading, and I do not think it is for the average user. In the end, we have Backend that supports important properties, why do we neglect them?

**Comment by herp1derp2** [comment]
> > these features are very important, and will speed up the downloading
> 
> I don’t doubt that they are, but what concrete use cases do they have? Why are they useful, and what problems do they solve? If they're useful they shouldn't be behind an "advanced" menu, and it's important to remember that ["advanced users" also want things to just work by default](https://www.joelonsoftware.com/2000/04/26/designing-for-people-who-have-better-things-to-do-with-their-lives/).

You are quite correct. Nobody really wants to fiddle inside settings if they don't have to. 
But like Apple do with their device and software, you can optimize your application in such a way that it takes advantage of all the backend software and projects you are using and the optimization and settings those softwares and projects offer so your users don't even have to go inside setting menu once. 

Having said that, I was originally here to ask if you'd be kind enough to give this wonderful and very useful piece of software the "tray icon" for Gnome, or some sort of indicator to make it run on background. Perhaps you can use Gnome's way of using XGD portal. 

Have a good one and thanks for making this software. It has been really useful. Cheers!

**Comment by giantpinkrobots** [comment]
> > > these features are very important, and will speed up the downloading
> > 
> > 
> > I don’t doubt that they are, but what concrete use cases do they have? Why are they useful, and what problems do they solve? If they're useful they shouldn't be behind an "advanced" menu, and it's important to remember that ["advanced users" also want things to just work by default](https://www.joelonsoftware.com/2000/04/26/designing-for-people-who-have-better-things-to-do-with-their-lives/).
> 
> You are quite correct. Nobody really wants to fiddle inside settings if they don't have to. But like Apple do with their device and software, you can optimize your application in such a way that it takes advantage of all the backend software and projects you are using and the optimization and settings those softwares and projects offer so your users don't even have to go inside setting menu even once.
> 
> H̶a̶v̶i̶n̶g̶ ̶s̶a̶i̶d̶ ̶t̶h̶a̶t̶,̶ ̶I̶ ̶w̶a̶s̶ ̶o̶r̶i̶g̶i̶n̶a̶l̶l̶y̶ ̶h̶e̶r̶e̶ ̶t̶o̶ ̶a̶s̶k̶ ̶i̶f̶ ̶y̶o̶u̶'̶d̶ ̶b̶e̶ ̶k̶i̶n̶d̶ ̶e̶n̶o̶u̶g̶h̶ ̶t̶o̶ ̶g̶i̶v̶e̶ ̶t̶h̶i̶s̶ ̶w̶o̶n̶d̶e̶r̶f̶u̶l̶ ̶a̶n̶d̶ ̶v̶e̶r̶y̶ ̶u̶s̶e̶f̶u̶l̶ ̶p̶i̶e̶c̶e̶ ̶o̶f̶ ̶s̶o̶f̶t̶w̶a̶r̶e̶ ̶t̶h̶e̶ ̶"̶t̶r̶a̶y̶ ̶i̶c̶o̶n̶"̶ ̶f̶o̶r̶ ̶G̶n̶o̶m̶e̶,̶ ̶o̶r̶ ̶s̶o̶m̶e̶ ̶s̶o̶r̶t̶ ̶o̶f̶ ̶i̶n̶d̶i̶c̶a̶t̶o̶r̶ ̶t̶o̶ ̶m̶a̶k̶e̶ ̶i̶t̶ ̶r̶u̶n̶ ̶o̶n̶ ̶b̶a̶c̶k̶g̶r̶o̶u̶n̶d̶.̶ ̶P̶e̶r̶h̶a̶p̶s̶ ̶y̶o̶u̶ ̶c̶a̶n̶ ̶u̶s̶e̶ ̶G̶n̶o̶m̶e̶'̶s̶ ̶w̶a̶y̶ ̶o̶f̶ ̶u̶s̶i̶n̶g̶ ̶X̶G̶D̶ ̶p̶o̶r̶t̶a̶l̶.̶ ̶
> 
> **Oh shid, nevermind, it is already on Gnome's background process and on their quick settings menu.**
> 
> Have a good one and thanks for making this software. It has been really useful. Cheers!

Wait no, it's not supposed to be on the Background Apps section in Gnome :/ that's a bug, there is no feature right now that allows Varia to run in the background. I have encountered this the other day but can't reproduce it consistently, maybe it's not shutting down aria2c properly.

Gotta go create an issue for it

**Comment by latenightdef** [comment]
I'm downloading an ISO file over a broadband connection. The download speed in Varia is just 3-4 MB/s while running `aria2c -x16 <link>` goes up to 7-8 MB/s. I think Varia should support this feature too! If a complete preferences pane is too much work, maybe we should start with a text field allowing the user to set custom aria2c command line flags?

**Comment by bragefuglseth** [comment]
I'm not knowledgeable on the technical details of Varia, but ideally those flags would be set automatically depending on network state / file type to optimize for fast downloads. Leaving it to a preference would make it inaccessible for most people.
